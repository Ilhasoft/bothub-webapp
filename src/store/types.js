export default {
  SET_TOKEN: 'SET_TOKEN',
  OPEN_LOGIN_MODAL: 'OPEN_LOGIN_MODAL',
  CLOSE_LOGIN_MODAL: 'CLOSE_LOGIN_MODAL',
  SET_PROFILE: 'SET_PROFILE',
  SET_NICKNAME_AUTHENTICATED: 'SET_NICKNAME_AUTHENTICATED',
  SET_REPOSITORY_RELATED_UUID: 'SET_REPOSITORY_RELATED_UUID',
  CACHE_FETCH: 'CACHE_FETCH',
};
