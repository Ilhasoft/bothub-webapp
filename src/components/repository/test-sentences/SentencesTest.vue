<template>
  <div class="bh-grid bh-grid--column">
    <div class="bh-grid">
      <h1>Test Sentences</h1>
    </div>
    <filter-sentences-test
      :intents="repository.intents_list"
      :labels="repository.labels_list"
      :entities="repository.entities_list"
      @queryStringFormated="onSearch($event)"
    />
    <sentences-test-list
      :repository="repository"
      :query="query"/>
  </div>
</template>

<script>
import Pagination from '@/components-v1/shared/Pagination';
import FilterSentencesTest from '@/components/repository/test-sentences/sentences/FilterSentencesTest';
import SentencesTestList from '@/components/repository/test-sentences/sentences/SentencesTestList';
import { exampleSearchToDicty, exampleSearchToString } from '@/utils/index';

export default {
  name: 'SentencesTest',
  components: {
    Pagination,
    FilterSentencesTest,
    SentencesTestList,
  },
  props: {
    repository: {
      type: Object,
      required: true,
    },
  },
  data() {
    return {
      querySchema: {},
      query: {},
    };
  },
  async mounted() {
    await this.updateList();
  },
  methods: {
    async updateList() {
      this.translateList = null;
      if (!!this.from && !!this.to) {
        await this.$nextTick();
        this.translateList = await this.getExamplesToTranslate({
          repositoryUuid: this.repository.uuid,
          from: this.from,
          to: this.to,
        });
      }
    },
    onSearch(value) {
      Object.assign(this.querySchema, value);

      if (!this.querySchema.intent) {
        delete this.querySchema.intent;
      }
      if (!this.querySchema.entitie) {
        delete this.querySchema.entitie;
      }
      if (!this.querySchema.label) {
        delete this.querySchema.label;
      }
      const formatedQueryString = exampleSearchToString(this.querySchema);
      this.query = exampleSearchToDicty(formatedQueryString);
    },
  },
};
</script>
