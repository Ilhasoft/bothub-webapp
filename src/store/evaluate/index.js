import actions from './actions';
import state from './state';
import mutations from './mutations';
import getters from './getters';
import types from './types';

export default {
  actions,
  state,
  mutations,
  getters,
  types,
};
