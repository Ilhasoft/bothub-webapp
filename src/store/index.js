import Vue from 'vue';
import Vuex from 'vuex';

import Auth from './auth';
import User from './user';
import Org from './org';
import Repository from './repository';
import Category from './category';
import Example from './example';
import EvaluateExample from './evaluate-example';
import Translate from './translate';
import CachedFetch from './cached-fetch';
import Entity from './entity';
import Intent from './intent';
import News from './news';
import Tutorial from './tutorial';
import External from './external';

Vue.use(Vuex);

const store = new Vuex.Store({
  modules: {
    Auth,
    User,
    Org,
    Repository,
    Category,
    Example,
    Translate,
    CachedFetch,
    EvaluateExample,
    Entity,
    News,
    Tutorial,
    External,
    Intent,
  },
});

store.dispatch('retriveAuthToken');

export default store;
