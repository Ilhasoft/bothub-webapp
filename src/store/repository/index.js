import state from './state';
import getters from './getters';
import actions from './actions';
import mutations from './mutations';
import types from './types';

export default {
  state,
  getters,
  actions,
  mutations,
  types,
};
