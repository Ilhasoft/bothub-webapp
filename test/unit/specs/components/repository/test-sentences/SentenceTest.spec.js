import { shallowMount, createLocalVue } from '@vue/test-utils';
import SentencesTest from '@/components/repository/test-sentences/SentencesTest';
import store from '@/store';
import BH from 'bh';


const localVue = createLocalVue();
localVue.use(BH);

describe('SentencesTest.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(SentencesTest, {
      localVue,
      store,
      propsData: {
        repository: {
          intents_list: [],
          labels_list: [],
          entities_lis: [],
        },
      },
    });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('Test filter query', () => {
    beforeEach(() => {
      wrapper.vm.onSearch({
        search: 'i like eat sushi',
        intent: 'like',
      });
    });

    test('Valid filter', () => {
      expect(wrapper.vm.query).toMatchObject({
        search: 'i like eat sushi',
        intent: 'like',
      });
    });

    test('Invalid filter', () => {
      expect(wrapper.vm.query).not.toMatchObject({
        search: 'i like eat sushi',
        intent: '',
      });
    });
  });
});
