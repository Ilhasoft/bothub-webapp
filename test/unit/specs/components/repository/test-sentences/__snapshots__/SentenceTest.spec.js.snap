// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SentencesTest.vue renders correctly 1`] = `
<div class="bh-grid bh-grid--column">
  <div class="bh-grid">
    <h1>Test Sentences</h1>
  </div>
  <filter-sentences-test-stub debouncetime="750" intents="" labels=""></filter-sentences-test-stub>
  <sentences-test-list-stub repository="[object Object]" query="[object Object]"></sentences-test-list-stub>
</div>
`;
